src/components/instructor/AssignmentUpdate.js

```diff
@@ -1,17 +1,93 @@
 import React, { useState } from 'react';
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle, TextField } from '@mui/material';
+import { SERVER_URL } from '../../Constants';
+
 
 //  instructor updates assignment title, dueDate 
 //  use an mui Dialog
 //  issue PUT to URL  /assignments with updated assignment
 
-const AssignmentUpdate = (props)  => {
 
+const AssignmentUpdate = (props) => {
+    const { assignment, save } = props;
+    const [open, setOpen] = useState(false);
+    const [title, setTitle] = useState(assignment.title);
+    const [dueDate, setDueDate] = useState(assignment.dueDate);
+    const [message, setMessage] = useState('');
+
+    const handleClickOpen = () => {
+        setOpen(true);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+        setMessage('');
+    };
+
+    const handleSave = async () => {
+        const updatedAssignment = { ...assignment, title, dueDate };
+        try {
+            const response = await fetch(`${SERVER_URL}/assignments`, {
+                method: 'PUT',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(updatedAssignment),
+            });
+            if (response.ok) {
+                setMessage('Assignment updated successfully');
+                save(updatedAssignment); // Assuming save will update the parent component's state
```

Comment:
What does this mean 'assuming'? This should be tested, so assumptions shouldn't really come into play here,

Reply:
I believe that the handleSave function should be used instead because you already defined it in assignmentUpdate.js. Props in this case, would be an assignment object and with the implementation of the other files, you should be able to see what functions those objects can call.